releasenotes/notes/add-backend-v2-ce84c976fb13b038.yaml

```diff
@@ -0,0 +1,87 @@
+---
+features:
+  - |
+    Added a new version of the :class:`~qiskit.providers.Backend` interface,
+    :class:`~qiskit.providers.BackendV2`. This new version is a large change
+    from the previous version, :class:`~qiskit.providers.BackendV1` and
+    changes both the user access pattern for properties of the backend (like
+    number of qubits, etc) and how the backend represents its constraints
+    to the transpiler. The execution of circuits (via the
+    :meth:`~qiskit.providers.BackendV2.run` method) remains unchanged. With
+    a :class:`~qiskit.providers.BackendV2` backend instead of having a separate
+    :meth:`~qiskit.providers.BackendV1.configuration`,
+    :meth:`~qiskit.providers.BackendV1.properties`, and
+    :meth:`~qiskit.providers.BackendV1.defaults` methods that construct
+    :class:`~qiskit.providers.models.BackendConfiguration`, 
+    :class:`~qiskit.providers.models.BackendProperties`, and
+    :class:`~qiskit.providers.models.PulseDefaults` objects respectively,
+    like in the :class:`~qiskit.providers.BackendV1` interface, the attributes
+    contained in those output objects are accessible directly as attributes of
+    the :class:`~qiskit.providers.BackendV2` object. For example, to get the
+    number of qubits for a backend with :class:`~qiskit.providers.BackendV1`
+    you would do::
+
+      num_qubits = backend.configuration().n_qubits
+
+    while with :class:`~qiskit.providers.BackendV2` it is::
+
+      num_qubits = backend.num_qubits
+
+    The other change around this is that the number of attributes exposed in
+    the abstract :class:`~qiskit.providers.BackendV2` class is designed to be
+    a hardware/vendor agnostic set of the required or optional fields that the
+    rest of Qiskit can use today with any backend. Anything can be added
+    to a particular implementation of :class:`~qiskit.providers.BackendV2`
+    as a custom addition for that particular backend. But it's not something
+    that is expected to be exposed by all backends.
```

Comment:
It might be worth clarifying this somewhat, to say something like "sub-classes can add support for additional attributes beyond those defined by `BackendV2`, but these will not be expected to be supported universally throughout Qiskit."

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/da767e4dfe0b22e4a5570f8a7979cdf4d1e77797